src/agent/src/namespace.rs

```diff
@@ -226,31 +227,28 @@ mod tests {
         assert!(ns_pid.is_err());
     }
 
-    #[test]
-    fn test_namespace_type() {
-        let ipc = NamespaceType::Ipc;
-        assert_eq!("ipc", ipc.get());
-        assert_eq!(CloneFlags::CLONE_NEWIPC, ipc.get_flags());
-
-        let uts = NamespaceType::Uts;
-        assert_eq!("uts", uts.get());
-        assert_eq!(CloneFlags::CLONE_NEWUTS, uts.get_flags());
-
-        let pid = NamespaceType::Pid;
-        assert_eq!("pid", pid.get());
-        assert_eq!(CloneFlags::CLONE_NEWPID, pid.get_flags());
+    #[rstest]
+    #[case(NamespaceType::Ipc, "ipc", CloneFlags::CLONE_NEWIPC)]
+    #[case(NamespaceType::Uts, "uts", CloneFlags::CLONE_NEWUTS)]
+    #[case(NamespaceType::Pid, "pid", CloneFlags::CLONE_NEWPID)]
+    fn test_namespace_type(
+        #[case] ns_type: NamespaceType,
+        #[case] ns_name: &str,
+        #[case] ns_flag: CloneFlags,
+    ) {
+        assert_eq!(ns_name, ns_type.get());
+        assert_eq!(ns_flag, ns_type.get_flags());
     }
 
-    #[test]
+    #[rstest]
```

Comment:
Hi Chengyu - this is a genuine question, I'm not saying that you have done anything wrong, but in my own projects I only use `#[rstest]` if I'm using a feature from it e.g. `[case...]` and it doesn't then I leave it as `#[test]` and have a mix of them in my unit tests. I see in this commit you've switched all the test cases to use `#[rstest]` - was there a particular reason for this?

Reply:
Hey Steve. The reason for switching all the test cases to `#[rstest]` in this commit was to maintain consistency across the test suite.

Reply:
But I think it's acceptable to have a mix of unit tests. Therefore, I have reverted the unit tests for which rstest is not necessary.

Reply:
Great - thanks for making the changes